src/lib/arch/riscv/InstructionMetadata.cc

```diff
@@ -0,0 +1,237 @@
+#include "InstructionMetadata.hh"
+
+#include <cassert>
+#include <cstring>
+#include <iostream>
+
+namespace simeng {
+namespace arch {
+namespace riscv {
+
+InstructionMetadata::InstructionMetadata(const cs_insn& insn)
+    : id(insn.id),
+      opcode(insn.opcode),
+      implicitSourceCount(insn.detail->regs_read_count),
+      implicitDestinationCount(insn.detail->regs_write_count),
+      operandCount(insn.detail->riscv.op_count) {
+  std::memcpy(encoding, insn.bytes, sizeof(encoding));
+  // Copy printed output
+  std::strncpy(mnemonic, insn.mnemonic, CS_MNEMONIC_SIZE);
+  operandStr = std::string(insn.op_str);
+
+  // Copy register/group/operand information
+  std::memcpy(implicitSources, insn.detail->regs_read,
+              sizeof(uint16_t) * implicitSourceCount);
+  std::memcpy(implicitDestinations, insn.detail->regs_write,
+              sizeof(uint16_t) * implicitDestinationCount);
+  std::memcpy(operands, insn.detail->riscv.operands,
+              sizeof(cs_riscv_op) * operandCount);
+
+  revertAliasing(insn);
+}
+
+InstructionMetadata::InstructionMetadata(const uint8_t* invalidEncoding,
+                                         uint8_t bytes)
+    : id(RISCV_INS_INVALID),
+      opcode(Opcode::RISCV_INSTRUCTION_LIST_END),
+      implicitSourceCount(0),
+      implicitDestinationCount(0),
+      operandCount(0) {
+  assert(bytes <= sizeof(encoding));
+  std::memcpy(encoding, invalidEncoding, bytes);
+  mnemonic[0] = '\0';
+  operandStr[0] = '\0';
+}
+
+void InstructionMetadata::revertAliasing(const cs_insn& insn) {
+  // Check for pseudoinstructions and change operands given by capstone to be
+  // suitable for SimEng
+  switch (opcode) {
+    case Opcode::RISCV_ADDI: {
```

Comment:
For every case, you should add a comment explaining what the instruction aliasing is. Look at this function in AArch64 metadata file for reference